packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -353,6 +354,19 @@ class VMService {
     await _peer.done;
   }
 
+  /// The connected physical device instance.
+  Device _device;
+  set device(Device device) {
```

Comment:
If you pass the device to VMService.connect, then maybe you don't need this complicated setter, and you can set up the extension in the constructor with all the other extensions.

Reply:
Done